plugins/org.python.pydev/tests/org/python/pydev/editor/actions/PyFormatBlackTest.java

```diff
@@ -0,0 +1,110 @@
+package org.python.pydev.editor.actions;
+
+import org.eclipse.jface.text.Document;
+import org.python.pydev.ast.formatter.PyFormatter;
+import org.python.pydev.core.docutils.SyntaxErrorException;
+import org.python.pydev.core.formatter.FormatStd;
+import org.python.pydev.core.formatter.FormatStd.FormatterEnum;
+import org.python.pydev.shared_core.resource_stubs.AbstractIFileStub;
+
+import junit.framework.TestCase;
+
+public class PyFormatBlackTest extends TestCase {
+    private final class MockIFile extends AbstractIFileStub {
+        private final String ext;
+
+        private MockIFile(String ext) {
+            this.ext = ext;
+        }
+
+        @Override
+        public String getFileExtension() {
+            return ext;
+        }
+    }
+
+    static final String SRC = "from typing import (\n"
+            + " Union, Dict\n"
+            + "  )\n"
+            + "def a()->None: \n"
+            + " ...\n"
+            + "def b(_:Union[Dict[str,str],str])->int: \n"
+            + " ...\n"
+            + "d = [1, 2, 3]";
+
+    private static boolean DEBUG = false;
+
+    public static void main(String[] args) {
+        try {
+            PyFormatBlackTest n = new PyFormatBlackTest();
+            junit.textui.TestRunner.run(PyFormatBlackTest.class);
+        } catch (Throwable e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private FormatStd black;
+
+    private void checkFormat(String ext, String expected) {
+        try {
+            Document doc = new Document(SRC);
+            PyFormatter.formatAll(doc, new MockPyEdit(new MockIFile(ext)), true, black, false, true);
+            String formattedStr = doc.get();
+
+            assertEquals(expected, formattedStr);
+        } catch (SyntaxErrorException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * @see junit.framework.TestCase#setUp()
+     */
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        black = new FormatStd();
+        black.formatterStyle = FormatterEnum.BLACK;
+        black.searchBlackInInterpreter = false;
+
+        //TODO: get black location from config?
+        black.blackExecutableLocation = System.getenv("LOCALAPPDATA")
+                + "\\Programs\\Python\\Python37\\Scripts\\black.exe";
```

Comment:
Not sure the best way to get black executable when running unit tests

Reply:
This should be based on something from `org.python.pydev.core.TestDependent`.

The values there are computed when TestDependent is imported.

You can create a file custom for your machine by setting the environment variable "PYDEV_TEST_PLATFORM" to some value -- say: `zeckie` -- and then create a `TestDependent.zeckie.properties` file based on the values of one of the existing templates there.

Ideally it should be a new entry there with a default value computed based on the `TestDependent.PYTHON_INSTALL` (to do that, change the `static{}` block on `TestDependent` to fill the value if it wasn't loaded).